docs/developer_guide/add_a_dataset.md

```diff
@@ -1,44 +1,127 @@
-Source code for datasets lives under `ludwig/datasets/`.
+The [Ludwig Dataset Zoo](../../user_guide/datasets/dataset_zoo) is a corpus of various datasets from the web
+conveniently built into Ludwig.
 
-# Adding a new Dataset
+Ludwig datasets come with several conveniences including: managing credentials for downloading data from sites like
+Kaggle, merging multiple files into a single dataset, sharing data parsing code, and loading datasets directly into data
+frames that can also be plugged into Ludwig models.
 
-Override `ludwig.datasets.base_dataset.BaseDataset` and implement the following methods:
+Ludwig datasets are defined under `ludwig/datasets/`.
+Dataset mixins are defined under `ludwig/datasets/mixins/`.
+For an example of a dataset, see `ludwig/datasets/wmt15/`.
+
+# 1. Create a new directory for your dataset
+
+Create a new directory under `ludwig/datasets/<dataset_name>`, and create two files:
+
+- `__init__.py`
+- `config.yaml`
+
+For an example, see `ludwig/datasets/wmt15`.
+
+# 2. Define a class that inherits from `ludwig.datasets.base_dataset.BaseDataset`
+
+We'll want to implement a class that inherits from `ludwig.datasets.base_dataset.BaseDataset` and implement the
+following methods:
 
 ```python
 @abc.abstractmethod
 def download_raw_dataset(self):
+    """Download the file from config.download_urls and save the file(s) as self.raw_dataset_path."""
     raise NotImplementedError()
 
-
 @abc.abstractmethod
 def process_downloaded_dataset(self):
+    """Process the dataset into a dataframe and save it as processed_dataset_path."""
     raise NotImplementedError()
 
-
 @abc.abstractmethod
 def load_processed_dataset(self, split):
+    """Loads the processed data from processed_dataset_path into a Pandas DataFrame in memory.
+
+    :param split: Splits dataset along 'split' column if present.
```

Comment:
Clarify that the implementation of load_processed_dataset is responsible for splitting the data.

Docs should mention that `load_processed_dataset` returns a dataframe if no split, and a 3-tuple of train, val, test if split.

Also I think the split column should always have values 0: train, 1: validation, 2: test.